Add guides for returns and a couple of related how-to guides (#61)
solidusio/edgeguides: docs/how-tos/how_to_skip_rmas_in_returns.mdx

---
sidebar_position: 2
---

> # How to skip RMAs in returns

Reviewer: I like how specific this how-to is, because skipping RMAs is definitely the most common use case for customizing the eligibility validators. However, I'm wondering if this might prevent users with other use cases from finding this how-to. What do you think?

Author: Thanks for pointing that out. It also bugged me, but I didn't know how to fix it.

I've changed the title to "How to customize the default eligibility in returns: skipping RMAs". What do you think? Being such a common problem, I think it makes sense to keep it in the title for better visibility.

Reviewer: How about this one?
```How to customize return eligibility rules: skipping RMAs```

Author: I like it. Updated.

> We'll see how you can disable RMAs as a prerequisite to creating a customer
> return. Solidus' [built-in process for returns][returns] is mighty and
> flexible. Only that sometimes a store doesn't need such a feature-rich system.

Reviewer: Given that the robustness and flexibility of the returns system is exactly what allows you to enable/disable RMAs so easily, I don't know if the last two sentences have a strong logical connection.

How about reframing this in a more positive way?
```suggestion
Solidus' [built-in process for returns][returns] is mighty and flexible, and
even allows you to disable RMAs as a prerequisite to creating a customer
return.
```

Author: Yeah, it makes sense. Integrated.

Go ahead and create an order from the storefront. You must go through

> everything, from adding items to the cart to confirming the payment.

Reviewer: ```suggestion
the entire checkout process, from adding items to the cart to confirming the payment.
```

Author: Integrated.

Next, you need to capture the payments in the system and perform the shipments.

> You can do it whether from the admin panel or through the console. E.g.:

Reviewer: ```suggestion
You can do it either from the admin panel or from the console. E.g.:
```

Author: Integrated.

```ruby
order = Spree::Order.find_by_number('R723438584')
order.payments.map(&:capture!)
order.shipments.map(&:ship!)
```

Let us go to the heart of the matter. When you add return items to a customer's

> return, Solidus checks whether they are eligible for that. By default, it uses

Reviewer: I find that, in English, being a bit more explicit often comes across as more natural than using implicit references (e.g., "for this", "for that"):
```suggestion
return, Solidus checks whether they are eligible for being returned. By default, it uses
```

Author: Thanks. Integrated.

[`Spree::ReturnItem::EligibilityValidator::Default`][eligibility-validator-default].
Even if you can [replace it altogether][configure-eligibility-validator], the
default validator is flexible enough to allow us to skip only the requirement
for an RMA.

Take a look at the stack of steps that `EligibilityValidator::Default` calls:
  
```ruby title="core/app/models/spree/return_item/eligibility_validator/default.rb"
# ...
self.permitted_eligibility_validators = [
  ReturnItem::EligibilityValidator::OrderCompleted,
  ReturnItem::EligibilityValidator::TimeSincePurchase,
  ReturnItem::EligibilityValidator::RMARequired,
  ReturnItem::EligibilityValidator::InventoryShipped,
  ReturnItem::EligibilityValidator::NoReimbursements
]
# ...
```

> Let's use Solidus initializer to remove the unwanted one:

Reviewer: ```suggestion
Let's use the Solidus initializer to remove the unwanted one:
```

Author: Integrated.
